src/dash-table/components/Export/utils.tsx

```diff
@@ -0,0 +1,74 @@
+import * as R from 'ramda';
+import XLSX from 'xlsx';
+
+interface IMergeObject {
+    s: {r: number, c: number};
+    e: {r: number, c: number};
+}
+
+export function transformMultDimArray(array: (string | string[])[], maxLength: number): string[][] {
+    const newArray: string[][] = array.map(row => {
+        if (row instanceof Array && row.length < maxLength) {
+            return row.concat(Array(maxLength - row.length).fill(''));
+        }
+        if (maxLength === 0 || maxLength === 1) {
+            return [row];
+        }
+        if (row instanceof String || typeof(row) === 'string') {
+            return Array(maxLength).fill(row);
+        }
+        return row;
+    });
+    return newArray;
+}
+
+export function getMergeRanges(array: string[][]) {
+    let apiMergeArray: IMergeObject[] = [];
+    const iForEachOuter = R.addIndex<(string[]), void>(R.forEach);
+    const iForEachInner = R.addIndex<(string), void>(R.forEach);
+    iForEachOuter((row: string[], rIndex: number) => {
```

Comment:
This is fine. You can simplify the double loop's + force Ramda explicit typing implementation by using `traverse2` from `core/math/matrixZipMap` instead.